dom/events/scrolling/scrollend-event-fired-for-programmatic-scroll.html

```diff
@@ -119,20 +109,20 @@
   async function testScrollFn(testInfo, t) {
     await waitForCompositorCommit();
 
-    targetDiv.addEventListener("scrollend", onElementScrollEnd);
-    document.addEventListener("scrollend", onDocumentScrollEnd);
+    let scrollendPromise = null;
+    if (testInfo.target == document.scrollingElement) {
+      targetDiv.addEventListener("scrollend", onInvalidScrollEnd);
+      scrollendPromise = createScrollendPromiseForTarget(t, document, 700, true);
+    } else {
+      document.addEventListener("scrollend", onInvalidScrollEnd);
```

Comment:
It seems like we should be calling
```js
test.add_cleanup(() => {document.removeEventListener("scrollend", onInvalidScrollEnd);});
```
Or ideally having a helper function to do this for us. This would ensure that if we ran multiple tests we wouldn't accidentally flag the scroll of the document as an error when we intended to scroll the document and could run all of the variants of this test directly as subtests rather than needing variants.